File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max=0;
        int diff;
        int point1=0;
        int point2=1;
        while(point2 < prices.size())
        {
            diff=prices[point2]-prices[point1];
            if(diff<0){
                point1 += 1;
                point2 = point1 + 1;
                continue;
            }
            else{
                if(diff>max){
                    max = diff;
                }
            }
                point2 += 1;

        }
        return max;
       
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxx=0
        temp=0
        for i in range(1,(len(prices))):
            diff = prices[i] - prices[temp]
            if(diff < 0):
                temp = i
                i = temp+1
                continue
            if(maxx<diff):
                maxx=diff
        return maxx